scenario_player/runner.py

```diff
@@ -2,46 +2,259 @@
+    get_contracts_deployment_info,
+)
+from requests import HTTPError, Session
 from web3 import HTTPProvider, Web3
-from web3.contract import Contract
 
 from raiden.accounts import Account
-from raiden.constants import GAS_LIMIT_FOR_TOKEN_CONTRACT_CALL
+from raiden.constants import UINT256_MAX
+from raiden.network.proxies.custom_token import CustomToken
+from raiden.network.proxies.proxy_manager import ProxyManager, ProxyManagerMetadata
+from raiden.network.proxies.token_network_registry import TokenNetworkRegistry
+from raiden.network.proxies.user_deposit import UserDeposit
 from raiden.network.rpc.client import JSONRPCClient
-from raiden.utils.nursery import Janitor, Nursery
-from raiden.utils.typing import TransactionHash
+from raiden.settings import RAIDEN_CONTRACT_VERSION
+from raiden.utils.formatting import to_canonical_address
+from raiden.utils.nursery import Janitor
+from raiden.utils.typing import (
+    Address,
+    BlockNumber,
+    TokenAddress,
+    TokenAmount,
+    TokenNetworkAddress,
+    TokenNetworkRegistryAddress,
+    UserDepositAddress,
+)
 from scenario_player.constants import (
     API_URL_TOKEN_NETWORK_ADDRESS,
-    API_URL_TOKENS,
     MAX_RAIDEN_STARTUP_TIME,
-    NODE_ACCOUNT_BALANCE_FUND,
     NODE_ACCOUNT_BALANCE_MIN,
     OWN_ACCOUNT_BALANCE_MIN,
     RUN_NUMBER_FILENAME,
 )
 from scenario_player.definition import ScenarioDefinition
-from scenario_player.exceptions import ScenarioError, TokenRegistrationError
+from scenario_player.exceptions import ScenarioError
 from scenario_player.exceptions.legacy import TokenNetworkDiscoveryTimeout
 from scenario_player.node_support import NodeController, RaidenReleaseKeeper
-from scenario_player.services.utils.interface import ServiceInterface
-from scenario_player.utils import TimeOutHTTPAdapter, get_udc_and_token, wait_for_txs
-from scenario_player.utils.token import Token, UserDepositContract
+from scenario_player.utils import TimeOutHTTPAdapter
+from scenario_player.utils.configuration.settings import SettingsConfig, UDCSettingsConfig
+from scenario_player.utils.token import (
+    TokenDetails,
+    eth_maybe_transfer,
+    load_token_configuration_from_file,
+    save_token_configuration_to_file,
+    token_maybe_mint,
+    userdeposit_maybe_deposit,
+    userdeposit_maybe_increase_allowance,
+)
 
 if TYPE_CHECKING:
     from scenario_player.tasks.base import Task, TaskState
 
 log = structlog.get_logger(__name__)
 
 
+# The `mint` function checks for overflow of the total supply. Here we
+# have a large enough number for both scenario runs and the scenario
+# orchestration account capacity.
+NUMBER_OF_RUNS_BEFORE_OVERFLOW = 2 ** 64
+ORCHESTRATION_MAXIMUM_BALANCE = UINT256_MAX // NUMBER_OF_RUNS_BEFORE_OVERFLOW
+
+
+def is_udc_enabled(udc_settings: UDCSettingsConfig):
+    should_deposit_ud_token = udc_settings.token.deposit
+    return udc_settings.enable and should_deposit_ud_token
+
+
+def wait_for_nodes_to_be_ready(node_runners, session):
```

Comment:
I just moved this loop out, since we don't really need it in a separate greenlet:

https://github.com/raiden-network/scenario-player/blob/edd041e816776c141a1182f089ab6bb75a976e91/scenario_player/runner.py#L403-L416